crates/biome_graphql_parser/src/parser/definitions/object.rs

```diff
@@ -29,3 +32,27 @@ pub(crate) fn parse_object_type_definition(p: &mut GraphqlParser) -> ParsedSynta
 
     Present(m.complete(p, GRAPHQL_OBJECT_TYPE_DEFINITION))
 }
+
+#[inline]
+pub(crate) fn parse_object_type_extension(p: &mut GraphqlParser) -> ParsedSyntax {
+    let m = p.start();
+
+    p.bump(T![extend]);
```

Comment:
I have placed a check on the outer `parse_extension` function [here](https://github.com/vohoanglong0107/biome/blob/865a0978951292315aaea6e9b2b73bbc83d946f3/crates/biome_graphql_parser/src/parser/definitions/mod.rs#L101), so unless this function was called outside of `parse_extension` it should be fine. I think panic would even be preferred in this case, cause it will fail the test and signify that this function was called at a unintended location. What do you think?

Reply:
Hmm, I'm unsure how to handle this better because I understand the cons and pros of both approaches.
I see two ways:

1. Add `is_at_object_type_extension` and use it only in `parse_object_type_extension`. This brings some duplication because we already check the correct parser state beforehand. However, I tend to use this approach.

2. Write documentation stating that we must check extends and type on the call sites; otherwise, it can cause a panic.

Reply:
I have added documents to check for the tokens. Since this pattern is also used in some other places, I have also documented them as well.